core/src/main/java/org/apache/calcite/sql/dialect/MysqlSqlDialect.java

```diff
@@ -67,7 +84,7 @@
 
   /** Creates a MysqlSqlDialect. */
   public MysqlSqlDialect(Context context) {
-    super(context);
+    super(context.withDataTypeSystem(MYSQL_TYPE_SYSTEM));
```

Comment:
No need for this.

Reply:
This is needed because `SqlDialectFactoryImpl.create` creates dialect with its own context (without using DEFAULT context).

Reply:
The context seems to be set externally and dialects just use the given context which would come with the type system set; observe that this is not being done for other dialects, e.g., Postgres. I see that _ SqlDialectFactoryImpl.create_ sets a bunch of properties depending on the database product. It seems we should add the type system over there instead of overriding within each individual constructor?

Reply:
Ok make sense